code/modules/mob/living/living_defense.dm

```diff
@@ -720,10 +721,8 @@
 	//Take their lunch money
 	var/target_held_item = target.get_active_held_item()
 	var/append_message = weapon ? " with [weapon]" : ""
-	if(!is_type_in_typecache(target_held_item, GLOB.shove_disarming_types)) //It's too expensive we'll get caught
-		target_held_item = null
-
-	if(target_held_item && target.get_timed_status_effect_duration(/datum/status_effect/staggered))
+	// If it's in our typecache, they're staggered and it exists, disarm. If they're knocked down, disarm too.
+	if(target_held_item && target.get_timed_status_effect_duration(/datum/status_effect/staggered) && is_type_in_typecache(target_held_item, GLOB.shove_disarming_types) || target_held_item && target.IsKnockdown())
```

Comment:
A bit confused on this logic. What did you change?

Reply:
Normally, when you shove twice, if that person has an item in the shove_disarming_types typecache, it will cause them to drop that item. That is the only instance where a shove directly removes an item from someones hands. This was used as an early return if the object wasn't in the typecache.

Now we're looking for either the original logic (Which is no longer an early return, or we're looking for someone holding an object and they're currently knocked down. 

Actually, it would be better to check their laying position now that I think about. I don't believe lying down = knockdown.